packages/legend-data-cube/src/stores/core/DataCubeQuerySnapshotBuilderUtils.ts

```diff
@@ -121,3 +166,291 @@ export function _pruneExpandedPaths(
+              (toApply.parameters[0].parameters[0] as V1_AppliedFunction)
+                .function == 'toLower'
+            ) {
+              let conditionSnapshot = _buildFilterConditionSnapshot(
+                toApply.parameters[0],
+              )!;
+              conditionSnapshot.not = true;
+              filterConditionSnapshot.push(conditionSnapshot);
+            } else {
+              filterConditionSnapshot.push(
+                _buildFilterConditionSnapshot(toApply)!,
+              );
+            }
+          }
+        }
+      }
+      // else
+      // {
+      //   toApply.parameters.forEach((param) => stack.push(param));
+      // }
+      // if (vs instanceof V1_AppliedFunction && (vs.function === DataCubeQueryFilterGroupOperator.OR))
+      // {
+      //   filterSnapshot.groupOperator = DataCubeQueryFilterGroupOperator.OR;
+      // }
+      if (
+        toApply instanceof V1_AppliedFunction &&
+        (toApply.function === DataCubeQueryFilterGroupOperator.OR ||
+          toApply.function === DataCubeQueryFilterGroupOperator.AND)
+      ) {
+        filterConditionSnapshot.push(_buildFilterConditionSnapshot(toApply)!);
+      }
+    }
+  }
+  filterSnapshot.conditions = filterConditionSnapshot;
+  return filterSnapshot;
+}
+
+function _buildFilterConditionSnapshot(
+  af: V1_AppliedFunction,
+): DataCubeQuerySnapshotFilterCondition | undefined {
+  if (af.parameters[1] && af.parameters[1] instanceof V1_AppliedProperty) {
+    switch (af.function) {
+      case DataCubeEngineFilterOperator.EQUAL:
+        return new DataCubeQueryFilterOperation__EqualColumn().buildConditionSnapshot(
+          af,
+        );
+      case DataCubeEngineFilterOperator.GREATER_THAN:
+        return new DataCubeQueryFilterOperation__GreaterThanColumn().buildConditionSnapshot(
+          af,
+        );
+      case DataCubeEngineFilterOperator.GREATER_THAN_OR_EQUAL:
+        return new DataCubeQueryFilterOperation__GreaterThanOrEqualColumn().buildConditionSnapshot(
+          af,
+        );
+      case DataCubeEngineFilterOperator.LESS_THAN:
+        return new DataCubeQueryFilterOperation__LessThanColumn().buildConditionSnapshot(
+          af,
+        );
+      case DataCubeEngineFilterOperator.LESS_THAN_OR_EQUAL:
+        return new DataCubeQueryFilterOperation__LessThanOrEqualColumn().buildConditionSnapshot(
+          af,
+        );
+      case DataCubeEngineFilterOperator.NOT:
+        return _buildNotFilterConditionSnapshot(
+          af.parameters[0] as V1_AppliedFunction,
+        );
+      default:
+        return undefined;
+    }
+  } else if (
+    af.parameters[0] instanceof V1_AppliedFunction &&
+    af.parameters[0].function === DataCubeEngineFilterOperator.TO_LOWERCASE
+  ) {
+    return _buildCaseInsensitiveFilterConditionSnapshot(af);
+  } else {
+    switch (af.function) {
+      case DataCubeEngineFilterOperator.CONTAINS:
+        return new DataCubeQueryFilterOperation__Contain().buildConditionSnapshot(
```

Comment:
this doesn't look right. The whole point of having the code for each operator contained in a file is to modularize this logic, infact, core logic of the app should not even know about any particular operation.

Reply:
resolved